lms/services/grouping.py

```diff
@@ -86,16 +87,21 @@ def upsert_grouping_memberships(self, user: User, groups: List[Grouping]):
         :param user:  User the that belongs to the groups
         :param groups: List of groups the `user` belongs to
         """
-        for group in groups:
-            if membership := (
-                self._db.query(GroupingMembership)
-                .filter_by(grouping_id=group.id, user_id=user.id)
-                .one_or_none()
-            ):
-                membership.updated = func.now()
-                continue
-
-            group.memberships.append(GroupingMembership(grouping=group, user=user))
+
+        bulk_upsert(
+            self._db,
+            GroupingMembership,
+            [
+                {
+                    "grouping_id": group.id,
+                    "user_id": user.id,
+                    "updated": func.now(),
```

Comment:
As this is the only column being updated we explicitly pass the column and value.